spec/entities/base_spec.rb

```diff
@@ -113,103 +187,181 @@ class Child < KalibroClient::Entities::Base; end
     end
   end
 
-  describe 'save' do
-    context "when it is not persisted" do
-      context "when it doesn't have the method id=" do
-        before :each do
-          KalibroClient::Entities::Base.
-            expects(:request).
-            with('', {base: {}}, :post, '').returns({"base" => {'id' => 42, 'kalibro_errors' => []}})
+  shared_examples 'persistence method' do |method_name, http_method, has_id = true|
+    before :each do
+      subject.id = 42 if has_id
+    end
+
+    let(:url) { has_id ? ':id' : '' }
+    let(:params) { has_id ? has_entry(id: 42) : anything }
+
+    context 'when a record does not exist with given id' do
+      before :each do
+        subject.class.expects(:request).with(url, params, http_method, '').
+          raises(KalibroClient::Errors::RecordNotFound)
+      end
+
+      it 'should raise a RecordNotFound error' do
+        expect { subject.send(method_name) }.to raise_error(KalibroClient::Errors::RecordNotFound)
+      end
+    end
+
+    context 'when a server error is returned' do
+      before :each do
+        error = KalibroClient::Errors::RequestError.new(response: mock(status: 500))
+
+        subject.class.expects(:request).with(url, params, http_method, '').raises(error)
+      end
+
+      it 'should raise a RequestError error' do
+        expect { subject.send(method_name) }.to raise_error(KalibroClient::Errors::RequestError)
+      end
+    end
+
+    context 'when a regular kind of error is returned' do
+      before :each do
+        error = KalibroClient::Errors::RequestError.new(response: mock(status: 422, body: { 'errors' => errors }))
+
+        subject.class.expects(:request).with(url, params, http_method, '').raises(error)
+      end
+
+      context 'with a single error' do
+        let(:errors) { "error" }
+
+        it 'should set the kalibro_errors field' do
+          expect(subject.send(method_name)).to eq(false)
+          expect(subject.kalibro_errors).to eq([errors])
         end
+      end
 
-        it 'should make a request to save model with id returning false and an error' do
-          expect(subject.save).to be(false)
-          expect(subject.kalibro_errors[0]).to be_a(NoMethodError)
+      context 'with an array of errors' do
+        let(:errors) { ["error_1", "error_2"] }
+
+        it 'should set the kalibro_errors field' do
+          expect(subject.send(method_name)).to eq(false)
+          expect(subject.kalibro_errors).to eq(errors)
         end
       end
 
-      context 'when it has the method id=' do
-        before :each do
-          KalibroClient::Entities::Base.
-            expects(:request).
-            with('', {base: {}}, :post, '').returns({"base" => {'id' => 42, 'kalibro_errors' => []}})
-          KalibroClient::Entities::Base.any_instance.expects(:id=).with(42).returns(42)
+      context 'with no error message at all' do
+        let(:errors) { nil }
+
+        it 'should set the kalibro_errors field' do
```

Comment:
Using `should` for spec descriptions has been deprecated. Instead, `is expected to` fits better to the code.